packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin_Test.h

```diff
@@ -14,4 +14,7 @@
 /// Inject @c GIDSignIn for testing.
 - (instancetype)initWithSignIn:(GIDSignIn *)signIn NS_DESIGNATED_INITIALIZER;
 
+/// Sets a NSBundle that the plugin will use instead of `NSBundle mainBundle`.
+- (void)overrideMainBundle:(NSBundle *)bundle;
```

Comment:
Can we instead inject the properties into the `initWithSignIn:` like this?
```suggestion
- (instancetype)initWithSignIn:(GIDSignIn *)signIn withGoogleServiceProperties:(nullable NSDictionary<NSString*, id>)googleServiceProperties
NS_DESIGNATED_INITIALIZER;
```
Injecting the dictionary values will give more flexibility instead of needing to update the actual example app plist.

Reply:
Good point. I left the original initializer method because we want to be able to test three scenarios:
1. Load GoogleService-Info.plist from bundle (original initializer)
2. Set `googleServiceProperties` to `nil`
3. Set `googleServiceProperties` to a value